File: food-delivery-backend/models/order.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type OrderItem struct {
	MenuItemID string  `json:"menuItemId" bson:"menuItemId"`
	Name       string  `json:"name" bson:"name"`
	Quantity   int     `json:"quantity" bson:"quantity"`
	Price      float64 `json:"price" bson:"price"`
}

type Order struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	UserID      primitive.ObjectID `bson:"userId" json:"userId"`
	Items       []OrderItem        `bson:"items" json:"items"`
	Subtotal    float64            `bson:"subtotal" json:"subtotal"`
	Tax         float64            `bson:"tax" json:"tax"`
	DeliveryFee float64            `bson:"deliveryFee" json:"deliveryFee"`
	PromoCode   string             `bson:"promoCode,omitempty" json:"promoCode"`
	Discount    float64            `bson:"discount" json:"discount"`
	Total       float64            `bson:"total" json:"total"`
	Status      string             `bson:"status" json:"status"`
	CreatedAt   time.Time          `bson:"createdAt" json:"createdAt"`
}

File: food_delivery_ui/lib/models/order.dart
class Order {
  final String id;
  final String status;
  final double total;
  final DateTime createdAt;
  final List<OrderItem> items;

  Order({
    required this.id,
    required this.status,
    required this.total,
    required this.createdAt,
    required this.items,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json["id"],
      status: json["status"],
      total: (json["total"] as num).toDouble(),
      createdAt: DateTime.parse(json["createdAt"]),
      items: (json["items"] as List)
          .map((e) => OrderItem.fromJson(e))
          .toList(),
    );
  }
}

class OrderItem {
  final String name;
  final int quantity;
  final double price;

  OrderItem({
    required this.name,
    required this.quantity,
    required this.price,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      name: json["name"],
      quantity: json["quantity"],
      price: (json["price"] as num).toDouble(),
    );
  }
}